NeoForge FMLCommonSetupEvent error at game launch

### Mod loader

NeoForge

### Minecraft version

1.21.1

### Patchouli version

1.21-88

### Modloader version

NeoForge: 21.1.133

### Modpack info

https://gist.github.com/Huskago/13fe1d383784a9d071286a7b04cf83cd

### The latest.log file

https://gist.github.com/Huskago/2d13e5a89e900c1fda86f14334313199

### Issue description

Sometimes when I start the game, before I get to the main menu, I have a problem that prevents me from playing.

![Image](https://github.com/user-attachments/assets/8732dbad-27b6-4db5-bbc9-dff6e80b7923)

### Steps to reproduce

- Launch the game
- If this does not appear, restart the game

### Other information

Information on the others can be found in the custom modpack link

Same error

```
[263��2025 17:02:56.427] [Worker-Main-14/ERROR] [net.neoforged.fml.javafmlmod.FMLModContainer/]: Exception caught during firing event: null
	Index: 0
	Listeners:
		0: net.neoforged.bus.EventListenerFactory$onInitialize/0x0000022802559c00@57a258bd
java.lang.ExceptionInInitializerError
	at TRANSFORMER/patchouli@1.21-87-NEOFORGE/vazkii.patchouli.common.book.BookRegistry.init(BookRegistry.java:42)
	at TRANSFORMER/patchouli@1.21-87-NEOFORGE/vazkii.patchouli.neoforge.common.NeoForgeModInitializer.onInitialize(NeoForgeModInitializer.java:86)
	at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.EventBus.post(EventBus.java:350)
	at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.EventBus.post(EventBus.java:315)
	at MC-BOOTSTRAP/fml_loader@4.0.38/net.neoforged.fml.ModContainer.acceptEvent(ModContainer.java:158)
	at MC-BOOTSTRAP/fml_loader@4.0.38/net.neoforged.fml.ModLoader.lambda$dispatchParallelEvent$4(ModLoader.java:172)
	at MC-BOOTSTRAP/fml_loader@4.0.38/net.neoforged.fml.ModLoader.lambda$dispatchParallelTask$7(ModLoader.java:231)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:911)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1491)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:2073)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2035)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
Caused by: java.lang.IllegalStateException: There should be exactly one IXplatAbstractions implementation on the classpath. Found: []
	at TRANSFORMER/patchouli@1.21-87-NEOFORGE/vazkii.patchouli.xplat.IXplatAbstractions.find(IXplatAbstractions.java:51)
	at TRANSFORMER/patchouli@1.21-87-NEOFORGE/vazkii.patchouli.xplat.IXplatAbstractions.<clinit>(IXplatAbstractions.java:45)
	... 15 more
```

--- a/Xplat/src/main/java/vazkii/patchouli/xplat/IClientXplatAbstractions.java
+++ b/Xplat/src/main/java/vazkii/patchouli/xplat/IClientXplatAbstractions.java
@@ -20,7 +20,7 @@ public interface IClientXplatAbstractions {
 	IClientXplatAbstractions INSTANCE = find();
 
 	private static IClientXplatAbstractions find() {
-		var providers = ServiceLoader.load(IClientXplatAbstractions.class).stream().toList();
+		var providers = ServiceLoader.load(IClientXplatAbstractions.class, IClientXplatAbstractions.class.getClassLoader()).stream().toList();
 		if (providers.size() != 1) {
 			var names = providers.stream().map(p -> p.type().getName()).collect(Collectors.joining(",", "[", "]"));
 			throw new IllegalStateException("There should be exactly one IClientXplatAbstractions implementation on the classpath. Found: " + names);

--- a/Xplat/src/main/java/vazkii/patchouli/xplat/IXplatAbstractions.java
+++ b/Xplat/src/main/java/vazkii/patchouli/xplat/IXplatAbstractions.java
@@ -45,7 +45,7 @@ public interface IXplatAbstractions {
 	IXplatAbstractions INSTANCE = find();
 
 	private static IXplatAbstractions find() {
-		var providers = ServiceLoader.load(IXplatAbstractions.class).stream().toList();
+		var providers = ServiceLoader.load(IXplatAbstractions.class, IXplatAbstractions.class.getClassLoader()).stream().toList();
 		if (providers.size() != 1) {
 			var names = providers.stream().map(p -> p.type().getName()).collect(Collectors.joining(",", "[", "]"));
 			throw new IllegalStateException("There should be exactly one IXplatAbstractions implementation on the classpath. Found: " + names);

